### PR TITLE
Nightly release

### PR DESCRIPTION
**Pull request by mwestphal**
Add a new "Nightly" workflow that:

- Check if something was commited in the last 24h, if it is store the commit and run the build adtion with the commit
- Advance the nightly tag
- Remove previous nightly asset
- Publish new assets

### CODE REVIEW

**Comment by mwestphal** [comment]
Sucessful CI can be seen here:  https://github.com/f3d-app/f3d-superbuild/actions/runs/3668692557

Resulting nightly release can be found here: https://github.com/f3d-app/f3d/releases/tag/nightly

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        git push https://${{ secrets.SECRET_TOKEN }}@github.com/f3d-app/f3d.git nightly -f
+
+    - name: Delete previous nightly release assets Linux
+      if: matrix.os == 'ubuntu-latest'
+      uses: mknejp/delete-release-assets@v1
+      with:
+        token: ${{ secrets.SECRET_TOKEN }}
+        repository: f3d-app/f3d
+        tag: nightly
+        assets: "F3D-*-Linux*"
```
Comment (Meakk) [inline .github/workflows/nightly.yml:75]: Can you factorize these three steps? Maybe using just `F3D-*` for all OS is enough?

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        submodules: 'true'
+        lfs: 'true'
+
+    - name: Build and package F3D
+      uses: ./source/.github/actions/f3d-superbuild/
+      with:
+        f3d_version: ${{ needs.check_nightly.outputs.f3d_sha}}
+        os: ${{matrix.os}}
+
+    - name: Advance nightly tag (Ubuntu Only)
```
Comment (Meakk) [inline .github/workflows/nightly.yml:60]: What happen if the tag is created on Ubuntu after the `Publish nightly release` step is run on Windows or macOS?

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (mwestphal) [reply .github/workflows/nightly.yml:75]: I need remove specific assets, but I will be able to factorize using matrix config. This is planned for the next MR

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (mwestphal) [reply .github/workflows/nightly.yml:60]: That's ok. I means that for a few minutes, the assets and the tag will be desynchronised, which will happen anyway.

**Review by Meakk — APPROVED** [review]

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      if: matrix.config.os == 'ubuntu-latest'
+      working-directory: ${{github.workspace}}/build/superbuild/f3d/src
+      shell: bash
+      run: |
+        git tag nightly -f
+        git push https://${{ secrets.SECRET_TOKEN }}@github.com/f3d-app/f3d.git nightly -f
+
+    - name: Delete previous nightly release assets Linux
```
Comment (mwestphal) [inline .github/workflows/nightly.yml:73]: fix name

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (mwestphal) [reply .github/workflows/nightly.yml]: fixed after all

**Comment by mwestphal** [comment]
@Meakk please review again.

Sucessfull CI: https://github.com/f3d-app/f3d-superbuild/pull/27/checks

Result: https://github.com/f3d-app/f3d/releases/tag/nightly

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      with:
+        repository: f3d-app/f3d
+        fetch-depth: 0
+        lfs: 'false'
+
+    - name: check latest commit is less than a day
+      id: should_run
+      continue-on-error: true
+      run: |
+        test -n "$(git log --since=72h)" && echo "should_run=true" >> $GITHUB_OUTPUT || echo "should_run=false" >> $GITHUB_OUTPUT
```
Comment (Meakk) [inline .github/workflows/nightly.yml:30]: `72h`? That's not a day

**Review by Meakk — CHANGES_REQUESTED** [review]

**Review by Meakk — APPROVED** [review]

**Pull request merged**